bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -228,7 +218,7 @@ pub mod pallet {
 				symbol: metadata.symbol.into_inner(),
 				decimals: metadata.decimals,
 			};
-			Self::send(origin_location, command, fee)?;
+			Self::send(origin, command, fee)?;
```

Comment:
Yes this seems good. Then in the gateway contract we can authorize this origin (/AH/SystemFrontend) to register tokens.

Reply:
In V1 it's always `Gateway` to create the token. Does that imply you want to change the behavior to introduce agent for PNA in V2?

Would that cause trouble for the incompatible?

Reply:
`message.origin` is not an agent, its just the hash of the origin location.

What I am saying is the Gateway could just check `message.origin == hash(SystemFrontendLocation)` instead of `message.origin == hash(BridgeHubLocation)`

But this is just a thought. I need to think about it more.